Formula/gmic.rb

```diff
@@ -20,19 +20,26 @@ class Gmic < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "4e61637e97caf2fdd854f10c685ef304731a607a8a4e0161d49b0dcfeeac08ef"
   end
 
-  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
   depends_on "fftw"
-  depends_on "jpeg"
+  depends_on "jpeg-turbo"
   depends_on "libpng"
   depends_on "libtiff"
 
+  uses_from_macos "curl"
+  uses_from_macos "zlib"
+
   def install
-    system "cmake", *std_cmake_args,
-                    "-DENABLE_FFMPEG=OFF",
-                    "-DENABLE_OPENCV=OFF",
-                    "-DENABLE_OPENEXR=OFF",
-                    "-DENABLE_X=OFF"
-    system "make", "install"
+    ENV.deparallelize
+    args = %W[
+      USR=#{prefix}
+      OPENEXR_CFLAGS=
+      OPENEXR_LIBS=
+      X11_CFLAGS=-Dcimg_display=0
+      X11_LIBS=
```

Comment:
Didn't find an easy way to disable features via Makefile unlike previous CMake. Manually clearing some variables seemed to be only option. X11 needs `cimg_display=0` to avoid trying to compile some code.

OpenCV doesn't look to be used using `make opencv`.

On side note, the libraries are all `.so`. No logic in Makefile to override
```console
❯ ls /usr/local/Cellar/gmic/3.1.5/lib
libgmic.so  libgmic.so.3  libgmic.so.315
```
```make
ifeq ($(OS),Darwin)
	$(CXX) -shared -std=c++11 -pedantic -o libgmic.so.$(VERSION) libgmic.o $(LIBS)
else
	$(CXX) -shared -std=c++11 -pedantic -Wl,-soname,libgmic.so.$(VERSION1) -o libgmic.so.$(VERSION) libgmic.o $(LIBS)
endif
```

Reply:
If upstream make it difficult to disable dependencies, we maybe shouldn't try. Unless other package managers are doing the same, I guess.

Reply:
The Linux examples I looked at (Arch, Fedora) enable most features.

MacPorts does something similar to what I added https://github.com/macports/macports-ports/blob/master/science/gmic/Portfile#L113-L121 (not sure if I should add `-lpthread`).
```
        if {[variant_isset x11]} {
            set x11_cflags     "-Dcimg_display=1 [exec ${prefix}/bin/pkg-config --cflags x11]"
            set x11_libs       "[exec ${prefix}/bin/pkg-config --libs x11] -lpthread"
        } else {
            set x11_cflags     "-Dcimg_display=0"
            set x11_libs       "-lpthread"
        }
        build.args-append       X11_CFLAGS="${x11_cflags}" \
                                X11_LIBS="${x11_libs}"
```

They also have a `.dylib` patch https://github.com/macports/macports-ports/blob/master/science/gmic/files/patch-src-Makefile.diff